packages/core/src/stepFunctions/activation.ts

```diff
@@ -37,14 +37,15 @@ export async function activate(
     await registerStepFunctionCommands(extensionContext, awsContext, outputChannel)
     initializeCodeLens(extensionContext)
 
+    if (isASLFileOpen()) {
+        await startLspServer(extensionContext)
```

Comment:
this should also call `onDidOpenAslDoc?.dispose()`, else a duplicate LSP server will start. 
```suggestion
        await startLspServer(extensionContext)
        onDidOpenAslDoc.dispose()
```

this needs to line 52

Reply:
I see, would something like this make sense then:

```
let onDidOpenAslDoc: vscode.Disposable // eslint-disable-line prefer-const
    // PERFORMANCE: Start the LSP client/server _only_ when the first ASL document is opened.
    // eslint-disable-next-line prefer-const
    onDidOpenAslDoc = vscode.window.onDidChangeActiveTextEditor(async e => {
        if (e?.document && ASL_FORMATS.includes(e.document.languageId)) {
            await startLspServer(extensionContext, onDidOpenAslDoc)
        }
    }, undefined)
    extensionContext.subscriptions.push(onDidOpenAslDoc)

    if (isASLFileOpen()) {
        await startLspServer(extensionContext, onDidOpenAslDoc)
    }
```

And we then update the `startLspServer` method to accept an `vscode.Disposable` type

```
async function startLspServer(extensionContext: vscode.ExtensionContext, onDidOpenAslDoc: vscode.Disposable) {
    const perflog = new PerfLog('stepFunctions: start LSP client/server')
    await ASLLanguageClient.create(extensionContext)
    perflog.done()
    onDidOpenAslDoc?.dispose() // Handler should only run once.
}
```

Reply:
Can't you just do something like:
```typescript
// PERFORMANCE: Start the LSP client/server _only_ when the first ASL document is opened. // eslint-disable-next-line prefer-const
const onDidOpenAslDoc = vscode.window.onDidChangeActiveTextEditor(async e => {
    if (e?.document && ASL_FORMATS.includes(e.document.languageId)) {
        await startLspServer(extensionContext, onDidOpenAslDoc)
    }
}, undefined)
extensionContext.subscriptions.push(onDidOpenAslDoc)

if (isASLFileOpen()) {
    onDidOpenAslDoc.dispose()
    await startLspServer(extensionContext)
}
```

then if a step functions file is open on activation onDidOpenASLDoc is instantly disposed (since we don't need it anymore, the server is started). If a file is not open at start then the if condition never gets hit and it waits for an open file. Rather than passing onDidOpenAslDoc through to the server start

Reply:
`await startLspServer(extensionContext, onDidOpenAslDoc)`

Did you meant to do this?:

```
await startLspServer(extensionContext)
onDidOpenAslDoc.dispose()
```

Everything else makes sense to me.